tests/test_humandetector.py

```diff
@@ -0,0 +1,293 @@
+
+    def test_batch_detect_by_area_without_human(self):
+        """
+        Test batch human detection by area without human
+        """
+        detection = self.detector.detect(
+            images=[ImageForDetection(image=VLIMAGE_ONE_FACE, detectArea=AREA_WITHOUT_FACE)]
+        )
+        assert 1 == len(detection)
+        assert 0 == len(detection[0])
+
+    def test_batch_detect_by_area_with_human(self):
+        """
+        Test batch human detection by area with human
+        """
+        detection = self.detector.detect(images=[ImageForDetection(image=VLIMAGE_ONE_FACE, detectArea=GOOD_AREA)])
+        assert 1 == len(detection[0])
+        self.assertHumanDetection(detection[0], VLIMAGE_ONE_FACE)
+
+    def test_bad_area_detection(self):
+        """
+        Test detection of one human in area outside image
+        """
+        with pytest.raises(LunaSDKException) as exceptionInfo:
+            self.detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=OUTSIDE_AREA)
+        self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidRect)
+
+    def test_batch_detect_in_area_outside_image(self):
+        """
+        Test batch detection in area outside image
+        """
+        with pytest.raises(LunaSDKException) as exceptionInfo:
+            self.detector.detect(images=[ImageForDetection(image=VLIMAGE_ONE_FACE, detectArea=OUTSIDE_AREA)])
+        self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidRect)
+
+    @pytest.mark.skip("unstable")
+    def test_excessive_image_list_detection(self):
+        """
+        Test excessive image list detection
+        """
+        with pytest.raises(LunaSDKException) as exceptionInfo:
+            TestHumanDetector.defaultDetector.detect(images=[VLIMAGE_ONE_FACE] * 1000)
+        self.assertLunaVlError(exceptionInfo, LunaVLError.HighMemoryUsage)
+
+    def test_detect_one_invalid_rectangle(self):
+        """
+        Test detection of one human with an invalid rect
+        """
+        with pytest.raises(LunaSDKException) as exceptionInfo:
+            self.detector.detectOne(image=VLIMAGE_ONE_FACE, detectArea=Rect())
+        self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidRect)
+
+    def test_batch_detect_invalid_rectangle(self):
+        """
+        Test batch human detection with an invalid rect
+        """
+        with pytest.raises(LunaSDKException) as exceptionInfo:
+            self.detector.detect(images=[ImageForDetection(image=VLIMAGE_ONE_FACE, detectArea=Rect())])
+        self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidRect)
+
+    def test_match_detection_one_image(self):
+        """
+        Test match of values at different detections (detectOne and detect) with one image
+        """
+        for image in (VLIMAGE_ONE_FACE, VLIMAGE_SMALL):
+            detectOne = self.detector.detectOne(image=image, detectLandmarks=True)
+            batchDetect = self.detector.detect(images=[image] * 3, detectLandmarks=True)
```

Comment:
one_image?

Reply:
yes. Test check that results are equal to a detectOne result